### PR TITLE
 Final Stage

### PR DESCRIPTION
**Pull request by Masouma96**
- Followed all the project requirements.
- [Click](https://budget-app-siir.onrender.com/) To preview the Live version, and
[Here ](https://www.loom.com/share/c73b32966b904cbf9704d23d3b3318da)is a video for explaining the project. 😄

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,168 @@
...
+
+- **[Budget App]**
+- **[Creating Transactions]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Here](https://budget-app-siir.onrender.com)👈😊
+
```
Comment (tresorsawasawa) [inline README.md:70]: - [ ] Good job trying to deploy your application online as required for the project. However, I noticed that it is still impossible for the user to be registered using the online deployed link. I suggest you fix that

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -5,16 +5,13 @@
...
-     <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0/dist/css/bootstrap.min.css" rel="stylesheet" >
+    <script src="https://cdn.jsdelivr.net/npm/@popperjs/core@2.9.2/dist/umd/popper.min.js" ></script>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.1/dist/js/bootstrap.min.js" integrity="sha384-Atwg2Pkwv9vp0ygtn1JAojH0nYbwNJLPhwyoVbhoPwBhjQPR5VtM2+xf0Uwh9KtT" crossorigin="anonymous"></script>
     <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap-icons@1.10.2/font/bootstrap-icons.css" integrity="sha384-b6lVK+yci+bfDmaY1u0zE8YYJt0TZxLEAFyYSLHId4xoVvsrQu3INevFKo+Xir8e" crossorigin="anonymous">
-   <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js"></script>
-   <script src="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
     <%#Fonts Cdn Links %>
     <link href="https://fonts.cdnfonts.com/css/proxima-nova-2" rel="stylesheet">
   </head>
```
Comment (tresorsawasawa) [inline app/views/layouts/application.html.erb:17]: - [ ] Kindly ensure that boith the **Login** and **Sign_Up** pages follows the template design as required for the project. Below are the expected designs from the template

| Login Page | Signup Page |
| -------------- | ------------------ |
| Screenshot from 2023-03-01 09-24-36 |Screenshot from 2023-03-01 09-24-58 |

File: spec/views/categories/index_spec.rb [hunk 3]
```diff
@@ -28,4 +28,4 @@
   #     end
   #     expect(current_path).to match group_expenses_path(@category)
   # end
```
Comment (tresorsawasawa) [inline spec/views/categories/index_spec.rb]: - [ ] For clean code and best practices, kindly ensure that you remove all the commented and used code lines from your code base

File: spec/views/categories/index_spec.rb [hunk 4]
```diff
@@ -28,4 +28,4 @@
...
-end
\ No newline at end of file
+end
```
Comment (tresorsawasawa) [inline spec/views/categories/index_spec.rb]: Screenshot from 2023-03-01 09-30-22
- [ ] `spec/views/transactions/index_spec.rb` file: For clean code and best practices, kindly ensure that you remove all the commented and used code lines from your code base

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @Masouma864 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow Gitflow :heavy_check_mark: 

## Required changes :recycle: :recycle: 

- [ ] Kindly ensure that your video contains a highlight about a piece of the code you find interesting or you are proud of making.

- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ishimwezachee — APPROVED** [review]
Hi @Masouma864 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights
- Linters are passing ✅ 
- Project is running successfully ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**